lisa/wa_results_collector.py

```diff
@@ -571,18 +580,25 @@ def _get_extra_job_metrics(self, job_dir, workload):
         else:
             return pd.DataFrame()
 
+    @memoized
+    def _wa_get_kernel_version(self, wa_dir):
```

Comment:
WAResultCollector instances already has a `plat_info` attribute, so you should be able to use `self.plat_info['kernel']['version']` (it's already a `KernelVersion` object [1]). 

Since plat_info is currently optional, you can build it in `WAResultCollector.__init__` with something like that:
```
if plat_info is None:
        with open(os.path.join(wa_dir, '__meta', 'target_info.json')) as f:
            target_info = json.load(f)

        plat_info = PlatformInfo({
              'kernel': {
                     'version': KernelVersion(target_info['kernel_release']),
               },
        })
self.plat_info = plat_info
```     

[1] https://lisa-linux-integrated-system-analysis.readthedocs.io/en/master/target.html#lisa.platforms.platinfo.PlatformInfo

Reply:
That wouldn't completely work here: WAResultsCollector is created for several WA "kernels". The value needs to be per WA folder.

Reply:
interesting, that means that this `plat_info` parameter is somewhat useless. The strange thing is that it replaces a parameter that used to be something equivalent, so it's surprising that it does not make sense. I guess that will wait for a round of cleanup and documentation.